### ISSUE TITLE
Node tutorial says freed instance references become null instead of invalid

### ISSUE THREAD

**Issue by wareya** [godotengine/godot-docs#8245]
**Your Godot version:**

Godot 4

**Issue description:**

In the `Nodes and scene instances` tutorial, the following line:

> You can alternatively call `free()` to immediately destroy the node. You should do this with care as any reference to it will instantly become `null`.

Should say "invalid" instead of "null". Saying that a reference becomes null means that it contains the `null` value, which is not what happens.

**URL to the documentation page:**

https://docs.godotengine.org/en/stable/tutorials/scripting/nodes_and_scene_instances.html#creating-nodes

**Comment by ratala321** [comment]
This is the code to check if an instance is valid after being freed/queue_freed and it does check if reference becomes null.

```c++
bool VariantUtilityFunctions::is_instance_valid(const Variant &p_instance) {
	if (p_instance.get_type() != Variant::OBJECT) {
		return false;
	}
	return p_instance.get_validated_object() != nullptr;
}
```

**Comment by AThousandShips** [comment]
Indeed, it *checks* for it, but it doesn't clear it, the method `p_instance.get_validated_object()` returns `nullptr` if the instance was freed, but the stored pointer is not `null`

**Comment by ratala321** [comment]
I couldn't find the binded c++ method for `free()`. If it's implementation is similar to queue_free, it's only a matter of time before `flush_delete_queue()` call `memdelete()` on the object.

Does anybody knows where to find the c++ code for `free()` in the Engine? This way, we would be sure.

**Comment by AThousandShips** [comment]
The pointer isn't cleared, there's no code doing that in the engine, the description in the documentation is wrong
